labonneboite/web/contact_form/views.py

```diff
@@ -0,0 +1,315 @@
+from wtforms.widgets import CheckboxInput, ListWidget
+from labonneboite.web.admin.views.office_admin_update import CONTACT_MODES
+
+from labonneboite.common import mapping as mapping_util
+from labonneboite.common.models import OtherRecruiterMessage, RemoveRecruiterMessage, UpdateCoordinatesRecruiterMessage, UpdateJobsRecruiterMessage
+from labonneboite.common.models import Office
+from labonneboite.web.contact_form import forms, mail
+from labonneboite.web.utils import fix_csrf_session
+
+logger = logging.getLogger('main')
+
+contactFormBlueprint = Blueprint('contact_form', __name__)
+
+class NoOfficeFoundException(Exception):
+    pass
+
+JOB_CHOICES = [
+    ('lbb', 'Intéressé par les candidatures'),
+    ('lba', 'Métier ouvert à l\'alternance'),
+    ('hide', 'Supprimer ce métier'),
+]
+
+# Get form value from office.contact_mode text
+LABEL_TO_VALUE_CONTACT_MODE = { v:k for k, v in CONTACT_MODES.items() }
+
+@contactFormBlueprint.route('/verification-informations-entreprise/<siret>')
+def change_info_or_apply_for_job(siret):
+    """
+    Ask user if he wants to change company information or apply for a job,
+    in order to avoid the change_info page to be spammed so much by
+    people thinking they are actually applying for a job.
+    """
+    return render_template('contact_form/change_info_or_apply_for_job.html', siret=siret)
+
+
+@contactFormBlueprint.route('/postuler/<siret>')
+def apply_for_job(siret):
+    """
+    If user arrives here, it means we successfully avoided having him spam the
+    company modification form. Now we just have to explain him what is wrong.
+    """
+    return render_template('contact_form/apply_for_job.html', siret=siret)
+
+
+@contactFormBlueprint.route('/informations-entreprise', methods=['GET', 'POST'])
+def change_info():
+    """
+    Let a user fill a form to identify himself/herself in order to request changes
+    about an office.
+    """
+    fix_csrf_session()
+    form = forms.OfficeIdentificationForm()
+
+    # Siret information is present when coming from change_info_or_apply_for_job route.
+    # Apply it only if user has not typed anything else yet.
+    params = request.args.copy()
+    siret = params.get('siret')
+    if siret and not form.data['siret']:
+        form.siret.data = siret
+
+    if form.validate_on_submit():
+        office = Office.query.filter(Office.siret == form.siret.data).first()
+        if not office:
+            flash('Ce siret n\'est pas connu de notre service', 'error')
+        else:
+            params = {
+                "siret": form.data['siret'],
+                "last_name": form.data['last_name'],
+                "first_name": form.data['first_name'],
+                "phone": form.data['phone'],
+                "email": form.data['email'],
+            }
+            if recruiter_from_lba():
+                params.update({"origin":"labonnealternance"})
+
+            # FIXME DNRY all fields
```

Comment:
Comment faire ? `**dict(form)` ?

Reply:
j'avoue que je bloque un peu sur ce point...
La seule amélioration que j'ai trouvé c'est de faire `params = {key:form.data[key] for key in ['siret','last_name','first_name','phone','email']}` pour réduire la longueur du code mais c'est tout :/

Reply:
C'est déjà pas mal. Avec cette dernière amélioration, il y a encore du code dupliqué ?